### PR TITLE
sessionctx: fix comments warning in sessionctx

### PR DESCRIPTION
**Pull request by knarfeh**

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
+	// preparedStmtID is prepared statement of auto-incremented id
 	preparedStmtID uint32
 
 	// retry information
 	RetryInfo *RetryInfo
 	// Should be reset on transaction finished.
 	TxnCtx *TransactionContext
 
-	// following variables are special for current session
+	// Status ... following variables are special for current session
```
Comment (zhexuany) [inline sessionctx/variable/session.go]: Another way to comment in go is `Status //following variables are special for current session ``

Comment (knarfeh) [inline sessionctx/variable/session.go]: Updated.

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 // SessionVars is to handle user-defined or global variables in current session.
 type SessionVars struct {
-	// user-defined variables
+	// Users variables
 	Users map[string]string
-	// system variables
+	// Systems variables
 	Systems map[string]string
-	// prepared statement
+	// PreparedStmts is prepared statement
```
Comment (ngaut) [inline sessionctx/variable/session.go]: s/PreparedStmts is prepared statement/PreparedStmts store prepared statement/

File: sessionctx/variable/session.go [hunk 3]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	PreparedStmts        map[uint32]interface{}
 	PreparedStmtNameToID map[string]uint32
-	// prepared statement auto increment id
+	// preparedStmtID is prepared statement of auto-incremented id
```
Comment (ngaut) [inline sessionctx/variable/session.go]: s/preparedStmtID is prepared statement of auto-incremented id/preparedStmtID is id of prepared statement/

File: sessionctx/varsutil/varsutil.go [hunk 4]
```diff
@@ -141,7 +141,7 @@ func SetSessionSystemVar(vars *variable.SessionVars, name string, value types.Da
 	return nil
 }
 
-// For all tidb session variable options, we use "ON"/1 to turn on the options.
+// tidbOptOn could be used for all tidb session variable options, we use "ON"/1 to turn on the options.
```
Comment (ngaut) [inline sessionctx/varsutil/varsutil.go]: s/the options/those options/

**Comment by knarfeh** [comment]
Updated.

File: sessionctx/variable/session.go [hunk 5]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 // SessionVars is to handle user-defined or global variables in current session.
 type SessionVars struct {
-	// user-defined variables
+	// Users variables
 	Users map[string]string
-	// system variables
+	// Systems variables
 	Systems map[string]string
-	// prepared statement
+	// PreparedStmts store prepared statement.
```
Comment (shenli) [inline sessionctx/variable/session.go]: store -> stores

File: sessionctx/variable/session.go [hunk 6]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
 
 // SessionVars is to handle user-defined or global variables in current session.
 type SessionVars struct {
-	// user-defined variables
+	// Users variables
```
Comment (shenli) [inline sessionctx/variable/session.go]: Users are user defined variables.

File: sessionctx/variable/session.go [hunk 7]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
+	// preparedStmtID is id of prepared statement.
 	preparedStmtID uint32
 
 	// retry information
 	RetryInfo *RetryInfo
 	// Should be reset on transaction finished.
 	TxnCtx *TransactionContext
 
-	// following variables are special for current session
+	// Status uint16 // following variables are special for current session
```
Comment (shenli) [inline sessionctx/variable/session.go]: // following variables are special for current session
// Status is the status of the current session.

File: sessionctx/variable/session.go [hunk 8]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	Status           uint16
 	PrevLastInsertID uint64 // PrevLastInsertID is the last insert ID of previous statement.
 	LastInsertID     uint64 // LastInsertID is the auto-generated ID in the current statement.
 	InsertID         uint64 // InsertID is the given insert ID of an auto_increment column.
 
-	// Client capability
+	// ClientCapability uint32 // Client capability
```
Comment (shenli) [inline sessionctx/variable/session.go]: ClientCapability is client's capability.

File: sessionctx/variable/session.go [hunk 9]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	ClientCapability uint32
 
-	// Connection ID
+	// ConnectionID uint64 // Connection ID
```
Comment (shenli) [inline sessionctx/variable/session.go]: ConnectionID is the connection id of the current session.

File: sessionctx/variable/session.go [hunk 10]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	ConnectionID uint64
 
-	// Current user
+	// User string // Current user
```
Comment (shenli) [inline sessionctx/variable/session.go]: User is the username with which the session login.

File: sessionctx/variable/session.go [hunk 11]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	User string
 
-	// Current DB
+	// CurrentDB string // Current DB
```
Comment (shenli) [inline sessionctx/variable/session.go]: CurrentDB is the default database of this session.

File: sessionctx/variable/session.go [hunk 12]
```diff
@@ -97,40 +97,40 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	CurrentDB string
 
-	// Strict SQL mode
+	// StrictSQLMode bool // Strict SQL mode
```
Comment (shenli) [inline sessionctx/variable/session.go]: StrictSQLMode indicates if the session is in strict mode.

File: sessionctx/variable/session.go [hunk 13]
```diff
@@ -173,32 +173,32 @@ type SessionVars struct {
...
-	// The number of concurrent dist SQL scan worker.
+	// DistSQLScanConcurrency is the number of concurrent dist SQL scan worker.
 	DistSQLScanConcurrency int
 
-	// The number of concurrent index serial scan worker.
+	// IndexSerialScanConcurrency is the number of concurrent index serial scan worker.
 	IndexSerialScanConcurrency int
 
-	// Should we split insert data into multiple batches.
+	// BatchInsert bool // Should we split insert data into multiple batches.
```
Comment (shenli) [inline sessionctx/variable/session.go]: BatchInsert indicates if we should split insert data into multiple batches.

File: sessionctx/variable/session.go [hunk 14]
```diff
@@ -97,40 +97,41 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
 
 // SessionVars is to handle user-defined or global variables in current session.
 type SessionVars struct {
-	// user-defined variables
+	// Users are user defined variables
 	Users map[string]string
-	// system variables
+	// Systems variables
```
Comment (coocood) [inline sessionctx/variable/session.go]: Systems are system variables.

File: sessionctx/variable/session.go [hunk 15]
```diff
@@ -97,40 +97,41 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
+	// preparedStmtID is id of prepared statement.
 	preparedStmtID uint32
 
 	// retry information
 	RetryInfo *RetryInfo
 	// Should be reset on transaction finished.
 	TxnCtx *TransactionContext
 
-	// following variables are special for current session
+	// Status is the status of the current session.
```
Comment (coocood) [inline sessionctx/variable/session.go]: Just add a blank line and keep the original comment will do.
The original comment describes several fields below.

File: sessionctx/variable/session.go [hunk 16]
```diff
@@ -300,13 +301,14 @@ type TableDelta struct {
 // StatementContext contains variables for a statement.
 // It should be reset before executing a statement.
 type StatementContext struct {
-	/* Variables that are set before execution */
+	// InUpdateOrDeleteStmt bool
+	// Following variables that are set before execution
```
Comment (coocood) [inline sessionctx/variable/session.go]: Just add an empty line will do.

File: sessionctx/variable/session.go [hunk 17]
```diff
@@ -97,40 +97,41 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
 
 // SessionVars is to handle user-defined or global variables in current session.
 type SessionVars struct {
-	// user-defined variables
+	// Users are user defined variables
```
Comment (XuHuaiyu) [inline sessionctx/variable/session.go]: add `.` at the end of this comment.

File: sessionctx/variable/session.go [hunk 18]
```diff
@@ -97,40 +97,41 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
 	Users map[string]string
-	// system variables
+	// Systems are system variables
```
Comment (XuHuaiyu) [inline sessionctx/variable/session.go]: ditto

File: sessionctx/variable/sysvar.go [hunk 19]
```diff
@@ -38,10 +38,10 @@ type SysVar struct {
 	// Scope is for whether can be changed or not
 	Scope ScopeFlag
 
-	// Variable name
+	// Name string // SysVar variable name
```
Comment (XuHuaiyu) [inline sessionctx/variable/sysvar.go]: Name is the variable name.

File: sessionctx/variable/sysvar.go [hunk 20]
```diff
@@ -38,10 +38,10 @@ type SysVar struct {
...
 	Name string
 
-	// Variable value
+	// Value string // SysVar variable value
```
Comment (XuHuaiyu) [inline sessionctx/variable/sysvar.go]: Value is the variable value.

**Comment by ngaut** [comment]
PTAL @Wenting0905

File: sessionctx/variable/session.go [hunk 21]
```diff
@@ -97,40 +97,41 @@ func (tc *TransactionContext) UpdateDeltaForTable(tableID int64, delta int64, co
...
+	// PreparedStmts stores prepared statement.
 	PreparedStmts        map[uint32]interface{}
 	PreparedStmtNameToID map[string]uint32
-	// prepared statement auto increment id
+	// preparedStmtID is id of prepared statement.
 	preparedStmtID uint32
 
 	// retry information
 	RetryInfo *RetryInfo
 	// Should be reset on transaction finished.
```
Comment (Wenting0905) [inline sessionctx/variable/session.go]: what does this sentence mean? when transaction finished? or "transaction finished" refers to "finished/completed transaction"?

File: sessionctx/variable/session.go [hunk 22]
```diff
@@ -173,32 +174,32 @@ type SessionVars struct {
 	// SkipConstraintCheck is true when importing data.
 	SkipConstraintCheck bool
 
-	// SkipUTF8 check on input value.
+	// SkipUTF8Check check on input value.
 	SkipUTF8Check bool
 
 	// SkipDDLWait can be set to true to skip 2 lease wait after create/drop/truncate table, create/drop database.
```
Comment (Wenting0905) [inline sessionctx/variable/session.go]: after creating/dropping/truncating table, creating/dropping databse

File: sessionctx/variable/session.go [hunk 23]
```diff
@@ -173,32 +174,32 @@ type SessionVars struct {
...
 	// Then if there are multiple TiDB servers, the new table may not be available for other TiDB servers.
 	SkipDDLWait bool
 
-	// TiDBBuildStatsConcurrency is used to control statistics building concurrency.
+	// BuildStatsConcurrencyVar is used to control statistics building concurrency.
 	BuildStatsConcurrencyVar int
 
-	// The number of handles for a index lookup task in index double read executor.
+	// IndexLookupSize is the number of handles for a index lookup task in index double read executor.
```
Comment (Wenting0905) [inline sessionctx/variable/session.go]: an index lookup task

File: sessionctx/variable/session.go [hunk 24]
```diff
@@ -300,13 +301,14 @@ type TableDelta struct {
 // StatementContext contains variables for a statement.
 // It should be reset before executing a statement.
 type StatementContext struct {
-	/* Variables that are set before execution */
+	// Following variables that are set before execution
```
Comment (Wenting0905) [inline sessionctx/variable/session.go]: Set the following variables before execution

File: sessionctx/variable/session.go [hunk 25]
```diff
@@ -300,13 +301,14 @@ type TableDelta struct {
...
+
 	InUpdateOrDeleteStmt bool
 	IgnoreTruncate       bool
 	TruncateAsWarning    bool
 	InShowWarning        bool
 
-	/* Variables that changes during execution. */
+	// mu struct holds variables that changes during execution.
```
Comment (Wenting0905) [inline sessionctx/variable/session.go]: change

File: sessionctx/variable/session.go (hunk 21 above)
Comment (coocood) [inline sessionctx/variable/session.go]: @Wenting0905 
Let's focus on the modified comments.

**Comment by coocood** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**